prog/vm/nexus.rb

```diff
@@ -176,6 +176,7 @@ def clear_stack_storage_volumes
   def before_run
     when_destroy_set? do
       if strand.label != "destroy"
+        vm.update(tainted_at: Time.now) unless vm.tainted_at
```

Comment:
I suppose one advantage of a "purity flag removal" idea is that the code to remove it is branchless, i.e. multiple unguarded updates do not mislead.

But a way to mitigate this would be to put the branch into a special method.

Reply:
I may refactor it to `vm.this.where(tainted_at: nil).update(tainted_at: Time.now)` to make it branchless, if you believe this is not an anti-pattern